lib/solidus_segment/configuration.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module SolidusSegment
+  def self.configure
+    yield configuration
+  end
+
+  def self.configuration
+    @configuration ||= Configuration.new
+  end
+
+  class Configuration
+    attr_writer :segment_backend
+    attr_accessor :segment_write_key
+
+    def segment_backend
+      @segment_backend ||= if defined?(Segment::Analytics)
+                             Segment::Analytics.new(write_key: segment_write_key)
+                           end
```

Comment:
In my opinion, if the gem doesn't exist, the gem should raise the default exception, however, this is more readable:

```ruby
@segment_backend ||= Segment::Analytics.new(write_key: segment_write_key) if defined? Segment::Analytics
```

Reply:
I think that the default Rubocop convention is useful to handle  these cases:
-> favor modifier if/unless usage when you have a single-line body unless it exceeds the maximum line length (default is 80 chars)
Here using the modifier would make line 109 chars which is bit long, I would stick with the current style

Reply:
You are right, I've updated the code to raise an error with a message that asks to set a backend or install the [analytics-ruby](https://github.com/segmentio/analytics-ruby) gem.